x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/endpoint_details_tabs.tsx

```diff
@@ -24,53 +29,80 @@ interface EndpointDetailsTabs {
   content: JSX.Element;
 }
 
-const StyledEuiTabbedContent = styled(EuiTabbedContent)`
-  overflow: hidden;
-  padding-bottom: ${(props) => props.theme.eui.paddingSizes.xl};
-
-  > [role='tabpanel'] {
-    height: 100%;
-    padding-right: 12px;
-    overflow: hidden;
-    overflow-y: auto;
-    ::-webkit-scrollbar {
-      -webkit-appearance: none;
-      width: 4px;
-    }
-    ::-webkit-scrollbar-thumb {
-      border-radius: 2px;
-      background-color: rgba(0, 0, 0, 0.5);
-      -webkit-box-shadow: 0 0 1px rgba(255, 255, 255, 0.5);
-    }
-  }
-`;
-
 export const EndpointDetailsFlyoutTabs = memo(
-  ({ show, tabs }: { show: EndpointIndexUIQueryParams['show']; tabs: EndpointDetailsTabs[] }) => {
+  ({
+    hostname,
+    show,
+    tabs,
+  }: {
+    hostname?: string;
+    show: EndpointIndexUIQueryParams['show'];
+    tabs: EndpointDetailsTabs[];
+  }) => {
+    const dispatch = useDispatch<(action: EndpointAction) => void>();
+    const { pageSize } = useEndpointSelector(getActivityLogDataPaging);
     const [selectedTabId, setSelectedTabId] = useState<EndpointDetailsTabsId>(() => {
```

Comment:
The selected tab id should be retrieved via a selector and not kept in this component as private state. See my comment below about driving it via URL rather than component state/store action